### PR TITLE
PURCHASE-1178 - Adds context card

### PR DESCRIPTION
**Pull request by kierangillen**
- https://artsyproduct.atlassian.net/browse/PURCHASE-1178

- Adds `ContextCard`
- Fixes bug in `PartnerCard` when no icon 
- Updates buttons to use `longestText` prop

## Examples

Screen Shot 2019-07-11 at 2 19 33 PM

Screen Shot 2019-07-11 at 2 10 55 PM

Screen Shot 2019-07-11 at 1 55 41 PM

### CODE REVIEW

**Comment by alloy** [comment]
As [mentioned in Slack](https://artsy.slack.com/archives/C9YNS4X32/p1562765587438200?thread_ts=1562695609.412000&cid=C9YNS4X32), I think you should use the existing `ArtworkContextPartnerShow` type for now.

File: src/lib/Scenes/Artwork/Components/PartnerCard.tsx [hunk 1]
```diff
@@ -86,7 +86,7 @@ export class PartnerCard extends React.Component<Props, State> {
       return null
     }
     const { isFollowedChanging } = this.state
-    const imageUrl = partner.profile ? partner.profile.icon.url : null
+    const imageUrl = partner.profile && partner.profile.icon ? partner.profile.icon.url : null
```
Comment (sweir27) [inline src/lib/Scenes/Artwork/Components/PartnerCard.tsx]: You can also use the `get` util here, I believe!

File: src/lib/Scenes/Artwork/Components/ContextCard.tsx [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+          id
+          name
+          href
+          formattedStartDateTime
+          cover_image {
+            url
+          }
+        }
+      }
+      shows(size: 1) {
```
Comment (sweir27) [inline src/lib/Scenes/Artwork/Components/ContextCard.tsx]: Is there a reason we're not spreading on the types for `show` and `fair`? (`... on ArtworkContextFair` and `... on ArtworkContextPartnerShow`)?

File: src/lib/Scenes/Artwork/Components/PartnerCard.tsx (hunk 1 above)
Comment (kierangillen) [reply src/lib/Scenes/Artwork/Components/PartnerCard.tsx]: Personally would rather not add overhead but personally preference.

Comment (ashfurrow) [reply src/lib/Scenes/Artwork/Components/PartnerCard.tsx]: `get()` also doesn't provide any kind of type safety 😞 We'll get `strictNullChecks` turned on here eventually 😅

File: src/lib/Scenes/Artwork/Components/ContextCard.tsx (hunk 2 above)
Comment (kierangillen) [reply src/lib/Scenes/Artwork/Components/ContextCard.tsx]: The reason we're not for Show is the context doesn't have a `ArtworkContextPartnerShow` type yet. Can add it for `Fair`.

Comment (kierangillen) [reply src/lib/Scenes/Artwork/Components/ContextCard.tsx]: Curious why we need to spread it in the context though if we're already querying for the fair directly on the Artwork.

File: src/lib/Scenes/Artwork/Components/PartnerCard.tsx (hunk 1 above)
Comment (sweir27) [reply src/lib/Scenes/Artwork/Components/PartnerCard.tsx]: Ah, I didn't mean importing directly from lodash.

We built a [typesafe version](https://github.com/artsy/emission/blob/master/src/lib/utils/get.ts) of this function in reaction, and it was imported here in April. 😄

File: src/lib/Scenes/Artwork/Components/ContextCard.tsx (hunk 2 above)
Comment (sweir27) [reply src/lib/Scenes/Artwork/Components/ContextCard.tsx]: Gotcha on the `show`.

The way it's written right now you will _always_ be querying for `fair` on the artwork, but by spreading on the union type, you'll only fetch for the resource when it's part of the context that's returned.

File: src/lib/Scenes/Artwork/Components/PartnerCard.tsx (hunk 1 above)
Comment (ashfurrow) [reply src/lib/Scenes/Artwork/Components/PartnerCard.tsx]: Oh, super-cool!

File: src/lib/Scenes/Artwork/Components/ContextCard.tsx (hunk 2 above)
Comment (kierangillen) [reply src/lib/Scenes/Artwork/Components/ContextCard.tsx]: Cool just updated!

File: src/lib/Scenes/Artwork/Artwork.tsx [hunk 3]
```diff
@@ -236,11 +245,20 @@ export const ArtworkContainer = createFragmentContainer(Artwork, {
       manufacturer
       image_rights
 
+      # For ContextCard
+      context {
+        __typename
+      }
+      shows(size: 1) {
```
Comment (sweir27) [inline src/lib/Scenes/Artwork/Artwork.tsx]: It doesn't seem like you are using this anymore in here.

**Pull request merged**